mindmeld/active_learning/heuristics.py

```diff
@@ -0,0 +1,621 @@
+from abc import ABC, abstractmethod
+from typing import List, Dict
+from collections import Counter
+from scipy.stats import entropy as scipy_entropy
+import numpy as np
+
+from ..constants import (
+    ENTROPY_LOG_BASE,
+    TRAIN_LEVEL_DOMAIN,
+    TRAIN_LEVEL_INTENT,
+    ACTIVE_LEARNING_RANDOM_SEED,
+)
+
+
+def custom_reordering(confidences, do_rank: bool = True, sampling_size: int = None):
+    """
+    Args:
+        confidences: List of confidence scores from heuristic.
+        do_rank: If True, this function will return rank-sorted confidence scores.
+        sampling_size: Size of data being sampled in a turn.
+    """
+    if not do_rank:
+        assert isinstance(sampling_size, int) and sampling_size > 0, print(
+            type(sampling_size), sampling_size
+        )
+        if sampling_size >= len(confidences):
+            return np.arange(len(confidences))
```

Comment:
Unsure why we do this, it seems like there is not sorting going on here, just returning a list of indices

Reply:
The name of this numpy function is a bit misleading. It doesn't intended to sort values but rather to create an evenly spaced list (similar to `list(range(5)`): 

Some examples to clarify this function

```
np.arange(3)
array([0, 1, 2])
np.arange(3.0)
array([ 0.,  1.,  2.])
np.arange(3,7)
array([3, 4, 5, 6])
np.arange(3,7,2)
array([3, 5])
```

Reply:
Refactored the heuristics.py file and removed this function. The new implementation of EnsembleSampling has a more simple approach to tabulating rankings so this function is not needed/used.